tests/unittests/sources/test_ec2.py

```diff
@@ -855,7 +854,7 @@ def test_unknown_platform_with_strict_true(self, mocker, tmpdir):
         """Unknown platform data with strict_id true should return False."""
         uuid = "ab439480-72bf-11d3-91fc-b8aded755F9a"
         ds = self._setup_ds(
-            platform_data={"uuid": uuid, "uuid_source": "dmi", "serial": ""},
+            platform_data={"uuid": uuid, "serial": ""},
```

Comment:
Why do we remove the `uuid_source` key?

Reply:
Good question. I should document this better in the commit message.

Cloud-init was previously making the assumption that uuid and serial would match on ec2. This assumption was:

1) [not documented as a valid way to identify ec2](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/identify_ec2_instances.html)

2) proven invalid on ec2 by the `DMI_PRODUCT_SERIAL` and `DMI_PRODUCT_UUID` reported in https://github.com/canonical/cloud-init/issues/5105

3) used in the logic which warns about not running on the "real" ec2

Preserving this warning logic exactly as it was presents several challenges:

a) Risk of regression outside of our control: Since this logic relied upon undocumented behavior, AWS could change this at any point, which would break all cloud-init instances.

b) Risk of incorrect implementation: What format is the uuid and product serial actually in? We don't know. It's easy and safe to just swap the byteorder of the first segment of the uuid because this is documented, but matching the whole uuid is problematic because UUID formats may be presented as mixed encoding (partially little endian and partially big endian). To implement this behavior while fixing this bug we would have to make even more assumptions than before. I propose we stop assuming and if a cloud happens to implement the same as EC2 (minus the serial/product match), then we just don't emit that warning. It's simpler, it's safer, and I really don't think that it is a huge change. This is a "change in behavior", but the change is that the code more correctly identifies EC2 and would no longer emit a warning on valid ec2 instances, so I don't think that this would require omitting this change from SRU.

c) Implementing whatever assumptions we make in b) would require a manual byteswapping algorithm in posix shell, which is more shell poo that I'd rather not write & maintain if I can convince folks that this preserving this strict ec2 warning _exactly_ as it was isn't worth it.

Reply:
> Why do we remove the uuid_source key?

To answer this question: because after the changes which resulted from the above comment, that key was no longer used anywhere.